hack/update-vendor.sh

```diff
@@ -180,210 +142,98 @@ function add_generated_comments() {
+      | xargs -L 100 go mod edit -fmt
+  go mod edit -json \
+      | jq -r '.Replace[]? | select(.New.Path | startswith("'"${path_to_staging_k8s_io}"'")) | "-dropreplace \(.Old.Path)"' \
+      | xargs -L 100 go mod edit -fmt
+  # Re-add
+  kube::util::list_staging_repos \
+      | while read -r X; do echo "-require k8s.io/${X}@v0.0.0"; done \
+      | xargs -L 100 go mod edit -fmt
+  kube::util::list_staging_repos \
+      | while read -r X; do echo "-replace k8s.io/${X}=${path_to_staging_k8s_io}/${X}"; done \
+      | xargs -L 100 go mod edit -fmt
+}
 
-# Phase 1: ensure go.mod files for staging modules and main module
-
+# === Ensure root and staging go.mod files refer to each other using v0.0.0 and local path replaces
+kube::log::status "go.mod: update staging module references" >&11
+add_staging_replace_directives "./staging/src/k8s.io"
 for repo in $(kube::util::list_staging_repos); do
   (
     cd "staging/src/k8s.io/${repo}"
-
-    if [[ ! -f go.mod ]]; then
-      kube::log::status "go.mod: initialize ${repo}" >&11
-      rm -f Godeps/Godeps.json # remove before initializing, staging Godeps are not authoritative
-      go mod init "k8s.io/${repo}"
-      go mod edit -fmt
-    fi
+    add_staging_replace_directives ".."
   )
 done
 
-if [[ ! -f go.mod ]]; then
-  kube::log::status "go.mod: initialize k8s.io/kubernetes" >&11
-  go mod init "k8s.io/kubernetes"
-  rm -f Godeps/Godeps.json # remove after initializing
-fi
-
-
-# Phase 2: ensure staging repo require/replace directives
-
-kube::log::status "go.mod: update staging references" >&11
-# Prune
-go mod edit -json \
-    | jq -r '.Require[]? | select(.Version == "v0.0.0")                 | "-droprequire \(.Path)"' \
-    | xargs -L 100 go mod edit -fmt
-go mod edit -json \
-    | jq -r '.Replace[]? | select(.New.Path | startswith("./staging/")) | "-dropreplace \(.Old.Path)"' \
-    | xargs -L 100 go mod edit -fmt
-# Re-add
-kube::util::list_staging_repos \
-    | while read -r X; do echo "-require k8s.io/${X}@v0.0.0"; done \
-    | xargs -L 100 go mod edit -fmt
-kube::util::list_staging_repos \
-    | while read -r X; do echo "-replace k8s.io/${X}=./staging/src/k8s.io/${X}"; done \
-    | xargs -L 100 go mod edit -fmt
-
-
-# Phase 3: capture required (minimum) versions from all modules, and replaced (pinned) versions from the root module
-
-# pin referenced versions
-ensure_require_replace_directives_for_all_dependencies
-# resolves/expands references in the root go.mod (if needed)
-go mod tidy
-# pin expanded versions
-ensure_require_replace_directives_for_all_dependencies
-# group require/replace directives
-group_directives
+# === Ensure all root and staging modules are included in go.work
+kube::log::status "go.mod: go work use" >&11
+(
+  cd "${KUBE_ROOT}"
+  unset GOWORK
+  unset GOFLAGS
+  go work use .
+  for repo in $(kube::util::list_staging_repos); do
+    go work use "./staging/src/k8s.io/${repo}"
```

Comment:
This sort of overlaps with update-go-workspace.sh -  should they fold together?

Reply:
Yeah, probably. I'll poke at it

Reply:
completely dropped `update-go-workspace.sh`, this now rebuilds go.work if needed